discovery-provider/plugins/network-monitoring/src/content/queries.ts

```diff
@@ -350,73 +357,102 @@ export const getSecondary2WalletBatch = async (
     return walletBatch
 }
 
+
 export const savePrimaryUserResults = async (
     run_id: number,
     spid: number,
     results: { walletPublicKey: string, clock: number }[],
 ): Promise<number> => {
 
-    try {
-        await Promise.all(
-            results.map(async ({ walletPublicKey, clock }) => {
-                try {
-                    await sequelizeConn.query(`
-                        UPDATE network_monitoring_users
-                        SET primary_clock_value = :clock
-                        WHERE wallet = :walletPublicKey
-                        AND run_id = :run_id;
-                `, {
-                        type: QueryTypes.UPDATE,
-                        replacements: { run_id, walletPublicKey, clock },
-                    })
-                } catch (e) {
-                    console.log(`[${run_id}:${spid}] error saving clock value (${clock}) to ${walletPublicKey} - ${(e as Error).message}`)
-                    return
+    const miniBatchSize = 500
+    const count = results.length
+
+    let missedUsers = 0
+
+    // Break the batch of clock value into smaller mini-batches 
+    // to cause less errors and make debugging easier
+    for (let offset = 0; offset < count; offset += miniBatchSize) {
+
+        const end = (offset + miniBatchSize) >= results.length ? results.length - 1 : (offset + miniBatchSize)
+
+        const miniBatch = results.slice(offset, end)
+
+        try {
+            const queryStr = `UPDATE network_monitoring_users as nm_users
+                    SET primary_clock_value = tmp.clock::text::int
+                    FROM (
+                        VALUES 
+                            ${formatUserValues(run_id, miniBatch)}
+                    ) AS tmp(run_id, wallet, clock)
+                    WHERE nm_users.wallet = tmp.wallet::text
+                    AND nm_users.run_id::text = tmp.run_id::text;`
+
+            await retry(
+                async () => {
+                    await sequelizeConn.query(queryStr, { type: QueryTypes.UPDATE })
+                },
+                {
+                    retries: 3,
+                    factor: 2,
+                    randomize: true,
                 }
-            })
-        )
-    } catch (e) {
-        console.log(`[${run_id}:${spid}:saveUserResults] error saving batch - ${(e as Error).message}`)
-        return results.length
+            )
+        } catch (e) {
+            console.log(`[${run_id}:${spid}:saveUserResults] error saving batch - ${(e as Error).message}`)
+            missedUsers += end - offset
```

Comment:
wait shouldn't this just be missedUsers += miniBatchSize
isnt this overcounting by a lot?

Reply:
Not necessarily, `miniBatchSize` is always 500 but `end-offset` is 500 or less. If the original batch length is 750, the mini batches will be 500 and 250. `miniBatchSize` won't account for the 250.

Reply:
sure but if offset = 0, end = 1000, miniBatchSize = 100, then end-offset = 1000

i think this has to be smth like `Math.floor(end-offset, miniBatchSize)`?

Reply:
end will never be more than `offset + miniBatchSize`

Reply:
oh lol i misread that logic, my bad